Drakhtar/DialogController.cpp

```diff
@@ -0,0 +1,32 @@
+#include "DialogController.h"
+
+
+DialogController::DialogController(DialogScene * dialogScene) : dialogScene_(dialogScene)
+{
+}
+
+void DialogController::run(SDL_Event event)
+{
+	SDL_Point p = { event.motion.x, event.motion.y };
+
+	switch (event.type) {
+	case SDL_MOUSEBUTTONUP:
+		if (!endOfDialog_)
+			onClick(p);
+		break;
+	}
+}
+
+void DialogController::onClick(SDL_Point p)
+{
+	GameObject * dialogBlock = dialogScene_->getBlockSprite();
+
+	if (p.x >= dialogBlock->getRect().x && p.y >= dialogBlock->getRect().y &&
```

Comment:
This should use [`SDL_PointInRect`](https://wiki.libsdl.org/SDL_PointInRect) instead.